src/ParamPassing.hh

```diff
@@ -0,0 +1,147 @@
+                                const bool _isModifyAction = false);
+
+    /// \brief Initialize an sdf element description from the xml element
+    /// \param[in] _xml Pointer to xml element
+    /// \param[out] _errors Captures errors found during parsing
+    /// \return ElementPtr to the initialized element description,
+    /// nullptr if undefined/unknown sdf element
+    ElementPtr initElementDescription(const tinyxml2::XMLElement *_xml,
+                                      Errors &_errors);
+
+    /// \brief Handles individual actions of children in _childrenXml
+    /// \param[in] _childrenXml Pointer to xml element
+    /// \param[out] _elem The sdf element to apply actions
+    /// (i.e., add, modify, remove, replace)
+    /// \param[out] _errors Captures errors found during parsing
+    void handleIndividualChildActions(tinyxml2::XMLElement *_childrenXml,
+                                      ElementPtr _elem, Errors &_errors);
+
+    /// \brief Adds a new element to an element from the included model
+    /// \param[in] _childXml Pointer to the new element to add, which is a child
+    /// of //include/experimental:params
+    /// \param[out] _elem The element from the included model to add the new
+    /// element to
+    /// \param[out] _errors Captures errors found during parsing
+    void add(tinyxml2::XMLElement *_childXml, ElementPtr _elem,
+             Errors &_errors);
+
+    /// \brief Modifies the attributes of an element from the included model
+    /// \param[in] _xml Pointer to the xml element which contains the attributes
+    /// to be modified
+    /// \param[out] _elem The element from the included model to modify the
+    /// attributes to
+    /// \param[out] _errors Captures errors found during parsing
+    void modifyAttributes(tinyxml2::XMLElement *_xml,
+                         ElementPtr _elem, Errors &_errors);
+
+    /// \brief Modifies element values and/or attributes of an element from the
+    /// included model
+    /// \param[in] _xml Pointer to the xml element which contains the elements
+    /// to be modified
+    /// \param[out] _elem The element from the included model to modify
+    /// \param[out] _errors Captures errors found during parsing
+    void modify(tinyxml2::XMLElement *_xml, ElementPtr _elem, Errors &_errors);
+
+    /// \brief Removes an element specified in xml
+    /// \param[in] _xml Pointer to the xml element(s) to be removed from _elem
+    /// \param[out] _elem The element from the included model to remove
+    /// elements from
+    /// \param[out] _errors Captures errors found during parsing
+    void remove(const tinyxml2::XMLElement *_xml, ElementPtr _elem,
+                Errors &_errors);
+
+    /// \brief Replace an element with another element
+    /// \param[in] _newElem The replacement element
+    /// \param[out] _origElem The element to be replaced
+    void replace(const ElementPtr _newElem, ElementPtr _origElem);
```

Comment:
`replace` doesn't emit errors since the errors differ when the `action` is defined in the `element_id` or if defined in the children elements. 

This is the error that occurs when the `action` is defined with `element_id`: 
https://github.com/osrf/sdformat/blob/81112de2c5fc18a071e8eacb9b9b2b2d3cfb9fae/src/ParamPassing.cc#L186-L195

These are the errors when the `action` is defined in the children of `element_id` (since there could be several children with different defined `action`s, this gave a clearer error for which child caused the issue and was skipped):
https://github.com/osrf/sdformat/blob/81112de2c5fc18a071e8eacb9b9b2b2d3cfb9fae/src/ParamPassing.cc#L479-L503